notebooks/04_model_select_and_optimize/hypertune_spark_deep_dive.ipynb

```diff
@@ -984,7 +929,7 @@
   },
```

Comment:
You use loguniform, not lognormal.
The following is not correct:
"In the ALS algorithm, the two hyper parameters, rank and reg, affect model performance in a way that\n",
    "* The higher the rank, the better the model performance but also the higher risk of overfitting.\n",
    "* The reg parameter is subject to rank and it prevents overfitting in certain way. \n",

Model performance (on test or validation data) will improve and then deteriorate, but it could go up and down more times. You don't know what will happen in advance. The reg. parameter prevents overfitting but is not related to the rank hyperparameter.